fluss-common/src/main/java/com/alibaba/fluss/cluster/Cluster.java

```diff
@@ -122,12 +122,19 @@ public Cluster(
         this.pathByTableId = Collections.unmodifiableMap(tempPathByTableId);
     }
 
-    public Cluster invalidPhysicalTableBucketMeta(
-            Collection<PhysicalTablePath> physicalTablesToInvalid) {
+    public Cluster invalidPhysicalTableBucketMeta(Set<PhysicalTablePath> physicalTablesToInvalid) {
+        // should remove invalid tables from current availableLocationsByPath
         Map<PhysicalTablePath, List<BucketLocation>> newBucketLocationsByPath =
-                new HashMap<>(availableLocationsByPath);
-        for (PhysicalTablePath path : physicalTablesToInvalid) {
-            newBucketLocationsByPath.remove(path);
+                new HashMap<>(availableLocationsByPath.size() - physicalTablesToInvalid.size());
```

Comment:
The the previous implementment: the value `List<BucketLocation>` is `UnmodifiableList`, new HashMap<>(availableLocationsByPath) won't unwrap the `UnmodifiableList` which will then cause we wrap `UnmodifiableList` again and again.. `StackOverflowError` will be thrown if the depth of wrapping overflows the length of stack...